src/sidebar/util/annotation-sharing.js

```diff
@@ -1,47 +1,61 @@
 /**
+ * @typedef {import('../../types/api').Annotation} Annotation
  * @typedef {import('../../types/config').HostConfig} HostConfig
  */
 
 import serviceConfig from '../service-config';
 
 /**
- * Is the sharing of annotations enabled? Check for any defined `serviceConfig`,
- * but default to `true` if none found.
+ * Generate a URI for sharing: a bouncer link built to share annotations in
+ * a specific group (groupID) on a specific document (documentURI).
  *
- * @param {HostConfig} settings
+ * @param {string} documentURI
+ * @param {string} groupID
+ * @return {string}
+ */
+export function getSharingLink(documentURI, groupID) {
+  return `https://hyp.is/go?url=${encodeURIComponent(
+    documentURI
+  )}&group=${groupID}`;
+}
+
+/**
+ * Are annotations made against `uri` meaningfully shareable? The
+ * target URI needs to be available on the web, which here is determined by
+ * a protocol of `http` or `https`.
+ *
+ * @param {string} uri
  * @return {boolean}
  */
-export function sharingEnabled(settings) {
-  const serviceConfig_ = serviceConfig(settings);
-  if (serviceConfig_ === null) {
-    return true;
-  }
-  if (typeof serviceConfig_.enableShareLinks !== 'boolean') {
-    return true;
-  }
-  return serviceConfig_.enableShareLinks;
+export function isShareableURI(uri) {
+  return /^http(s?):/.test(uri);
 }
 
 /**
- * Return any defined standalone URI for this `annotation`, preferably the
- * `incontext` URI, but fallback to `html` link if not present.
+ * Return the service-provided sharing URI for an annotation. Prefer the
+ * `incontext` link when available; fallback to `html` if not present.
  *
- * @param {object} annotation
+ * @param {Annotation} annotation
  * @return {string|undefined}
  */
 export function shareURI(annotation) {
-  const links = annotation.links;
-  return links && (links.incontext || links.html);
+  return annotation.links?.incontext ?? annotation.links?.html;
 }
 
 /**
- * For an annotation to be "shareable", sharing links need to be enabled overall
- * and the annotation itself needs to have a sharing URI.
+ * Is the sharing of annotations enabled? Check for any defined `serviceConfig`,
+ * but default to `true` if none found.
  *
- * @param {object} annotation
  * @param {HostConfig} settings
  * @return {boolean}
  */
-export function isShareable(annotation, settings) {
-  return !!(sharingEnabled(settings) && shareURI(annotation));
+export function sharingEnabled(settings) {
```

Comment:
Note that we check config (i.e. call this function) to determine whether it should be allowed to share single annotations, but the logic for showing the "share all annotations on this page" is simply "is this first-party?" (tested in `AnnotationActionBar`).